core/src/test/java/io/grpc/internal/BinaryLogProviderTest.java

```diff
@@ -50,6 +119,134 @@ public void unavailableProvider() {
     assertSame(BinaryLogProvider.NullProvider.class, BinaryLogProvider.load(cl).getClass());
   }
 
+  @Test
+  public void serverInterceptorFirstAndIsInputstream() throws Exception {
```

Comment:
This test is in the wrong test class. This is really just testing ServerImpl today, as "first" and "InputStream" are mostly decided by ServerImpl (yes, InputStream isn't, but _converting_ the type is, and forgetting to convert is much more likely than choosing the wrong type).

Granted, I've provided some suggestions that would shift that around a bit, but I think a lot of the infrastructure here would still move to ServerImplTest. And then you will have a simpler test here.

Reply:
Moved